lib/rubocop/ast/node/mixin/parameterized_trail_node.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module AST
+    # Same functionality as ParameterizedNode, but optimized for Nodes
+    # where `arguments` are `children[first_argument_index..-1]`
+    #
+    module ParameterizedTrailNode
```

Comment:
I don't quite get the "trail" terminology.

Reply:
The first few children are not the arguments, only the trailing ones.
I don't like `ParameterizedNode`, and I dislike `ParameterizedTrailNode` even more 😅, but I'm severly uninspired. Suggestions?

Reply:
@bbatsov is `TrailingArgumentsNode` a better name?

Reply:
ping @bbatsov

Reply:
It gets a bit hairy, I guess, because the naming of the other nodes make reference to Ruby and its syntax, while this is a reference to the Parser output.

Reply:
@Drenmi right, but there is still shared handling of the AST like in `WrappedArgumentsNode` (that isn't just an optimization)

One way would be to have `ParameterizedNode::SingleChildArguments` / `ParameterizedNode::LastChildrenArguments` and `ParameterizedNode::WrappedArguments`  or similar, to denote the 3 different ways `Parser` encodes arguments?

This could even be handled automatically by `ParameterizedNode` when there's a `def first_argument_index` or similar that is encountered, but it feels too much.